models/articles.model.js

```diff
@@ -3,12 +3,23 @@ exports.selectArticles = () => {
     const queryStr = `
         SELECT articles.author, title, articles.article_id, topic, articles.created_at, 
         articles.votes, COUNT(comments.article_id) as comment_count FROM articles
-        JOIN comments
+        LEFT JOIN comments
             ON articles.article_id = comments.article_id
         GROUP BY articles.article_id
         ORDER BY articles.article_id;
     `
     return db.query(queryStr).then((results) => {
         return results.rows
     })
+}
+exports.selectArticleById = (article_id) => { 
+    if(isNaN(parseInt(article_id)))  {
```

Comment:
If you pass a non-integer into the PSQL query, you'll get an error back from PSQL rather than having to write this yourself. Not bad logic, but definitely easier to handle the PSQL error.